[WilkinsAng] iP

WilkinsAng opened the pull request:
# Mona
This is Morgana, codename: Mona
> I'm not a cat!


Ever wanted a ~~cat~~ human to tell you to go to sleep? 
Look no further! Mona can do much more! It's

- Text Based
- Easy to learn
- **SUPER FAST** to use

All you have to do is

1. download it from [here](https://megamitensei.fandom.com/wiki/Morgana).
2. double-click it.
3. add your tasks. 
4. let the ~~cat~~ manage your tasks for you 🐈‍⬛ .
 And it is **FREE**!

Here is what he can do,

- [x] Manage your Tasks
- [x] Manage your Deadlines
- [x] Find your forgotten tasks
- [x] Remember and save ur tasks
- [ ] Lets you see his face
- [x] Tells you to go to sleep
<hr/>

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Mona {
    public static void main(String[] args) {
            new Mona("data/Mona.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

WilkinsAng committed:
Rename Bot, enable Greet, Exit

WilkinsAng committed:
Testing

WilkinsAng committed:
Improve the skeletal version of Duke so that it echos commands entered by the user, and exits when the user types the command bye.

WilkinsAng committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

WilkinsAng committed:
Add the ability to mark tasks as done and not done

WilkinsAng committed:
Add support for tracking three types of tasks: ToDos, Deadlines, Events

WilkinsAng committed:
Add Automated Text UI Testing

WilkinsAng committed:
Add MonaException for error handling

WilkinsAng committed:
Update MonaExceptions with more error handling

WilkinsAng committed:
Add support for Error Handling

WilkinsAng committed:
Add support for deleting tasks from the list

WilkinsAng committed:
Update Commands to use Enum

WilkinsAng committed:
Add JavaDocs for Task Class

WilkinsAng committed:
Move Enums for better OOP

WilkinsAng committed:
Add support to load a save

WilkinsAng committed:
Add support to reset file if file is corrupted

WilkinsAng committed:
Add saving formats for Tasks

WilkinsAng committed:
Add full support for save and load

WilkinsAng committed:
Update gitignore

WilkinsAng committed:
Merge branch 'branch-Level-7'

WilkinsAng committed:
Add new parent TimedTask class and new exception handling for dates

WilkinsAng committed:
Update deadline and event to save and print dates

WilkinsAng committed:
Add support for dates and time over strings and update tests

WilkinsAng committed:
Merge branch 'branch-Level-8'

J4Joshua commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,30 @@
+import java.time.LocalDateTime;
```
Could put all task classes in a task package

J4Joshua commented on src/main/java/Mona.java:
```diff
@@ -0,0 +1,192 @@
+import java.util.Scanner;
+
+
+public class Mona {
```
A brief description of Mona would be good

J4Joshua commented on src/main/java/Mona.java:
```diff
@@ -0,0 +1,192 @@
+            try {
+                String[] splitMsg = message.split(" ");
+                Command command = Command.fromString(splitMsg[0]);
+                switch (command) {
```
Good indentation for switch

J4Joshua reviewed (commented):
LGTM just split your classes into packages for easier management

WilkinsAng committed:
Add new class, TaskList, which handles operations on the list of tasks

WilkinsAng committed:
Add new classes for Command and parsing logic

WilkinsAng committed:
Refactor classes to be aligned with OOP principles

WilkinsAng committed:
Add Packages and refactored files

WilkinsAng committed:
Merge branch 'branch-A-Packages'

WilkinsAng committed:
Merge branch 'add-gradle-support'

WilkinsAng committed:
Update build.gradle to reflect Mona

WilkinsAng committed:
Add minimal JUnit Testing

WilkinsAng committed:
Add JavaDocs to Command package

WilkinsAng committed:
Refactor code in Commands to be more in line with coding conventions

WilkinsAng committed:
Add JavaDocs for exceptions

WilkinsAng committed:
Refactor Exceptions to be in line with coding standards

WilkinsAng committed:
Add javadocs for the parser class

WilkinsAng committed:
Refactor parser class to be in line with coding standards

WilkinsAng committed:
Add javadocs for storage class

WilkinsAng committed:
Refactor storage to be in line with coding standards

WilkinsAng committed:
Add javadocs to Task package

WilkinsAng committed:
Refactor Task packages to be in line with coding standard

WilkinsAng committed:
Add javadocs for UI package

WilkinsAng committed:
Refactor ui package to be in line with coding standards

WilkinsAng committed:
Added javadocs to Mona class

WilkinsAng committed:
Add support for find function

WilkinsAng committed:
Merge branch 'branch-A-JavaDoc'

WilkinsAng committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/command/DeleteCommand.java
#	src/main/java/command/ListCommand.java
#	src/main/java/command/MarkCommand.java
#	src/main/java/mona/command/Command.java
#	src/main/java/mona/command/UnmarkCommand.java
#	src/main/java/mona/exception/MonaException.java
#	src/main/java/mona/parser/Parser.java
#	src/main/java/mona/task/Task.java
#	src/main/java/mona/task/TaskList.java

WilkinsAng committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/mona/command/FindCommand.java
#	src/main/java/mona/exception/MonaException.java
#	src/main/java/mona/parser/Parser.java
#	src/main/java/mona/ui/Ui.java
#	src/main/java/task/TaskList.java

WilkinsAng committed:
Refactor code to make last 5 commits meet standards

WilkinsAng committed:
Add checkstyle to repo

WilkinsAng committed:
Merge branch 'branch-A-CheckStyle'

WilkinsAng committed:
Refector code to be in line with checkstyle

WilkinsAng committed:
Add JavaFX code and refactor existing code for GUI

WilkinsAng committed:
Update functionalities to GUI instead of CLI

WilkinsAng committed:
Add greeting to Mona GUI, finish minimal level 10

WilkinsAng committed:
Merge branch 'branch-Level-10'

WilkinsAng committed:
Add Varargs to Find function so that we can search for more than 1 query

WilkinsAng committed:
Merge branch 'branch-A-Varargs'

WilkinsAng committed:
Update Gradle for new JAR

WilkinsAng committed:
Add Assertions to most classes

WilkinsAng committed:
Merge branch 'master' into branch-A-Assertions

WilkinsAng committed:
Merge pull request #1 from WilkinsAng/branch-A-Assertions

Add Assertions to classes

WilkinsAng committed:
Merge branch 'master' into branch-A-CodeQuality

WilkinsAng committed:
Add Parser for storage to make loading method shorter

WilkinsAng committed:
Update Loading files to be in line with SLAP and add GUI support for loading/saving functions

WilkinsAng committed:
Merge pull request #5 from WilkinsAng/branch-A-CodeQuality

Update Code for Code Quality

WilkinsAng committed:
Merge branch 'master' into branch-A-Streams

WilkinsAng committed:
Update find function to use Streams

Find function uses loops to search through queries and the task list

Let's update the format to use streams

So that it is
* easier to read
* more concise

WilkinsAng committed:
Update function to list tasks

Show all tasks method uses Stringbuilder

A stream format is more reading-friendly because it is more concise
and does not have so many append calls.

Let's
* update the method to incorporate streams

WilkinsAng committed:
Update find display function

Show found tasks method uses Stringbuilder

A stream format is more reading-friendly because it is more concise
and does not have so many append calls.

Let's
* update the method to incorporate streams

WilkinsAng committed:
Merge branch 'branch-A-Streams'

WilkinsAng committed:
Merge pull request #6 from WilkinsAng/branch-A-Streams

Refactor code to use Streams

WilkinsAng committed:
Merge branch 'master' of https://github.com/WilkinsAng/ip

WilkinsAng committed:
Add support for CI

WilkinsAng committed:
Merge branch 'master' into branch-HandleLoadingSavingErrors

WilkinsAng committed:
Update Test files for CI

WilkinsAng committed:
Merge branch 'branch-HandleLoadingSavingErrors'

WilkinsAng committed:
Add new Task Priority Class

WilkinsAng committed:
Update task classes for priority support

WilkinsAng committed:
Update task classes to save priority

WilkinsAng committed:
Update storage to support priority

WilkinsAng committed:
Add support for prioritize function

WilkinsAng committed:
Merge branch 'branch-C-Priority'

WilkinsAng committed:
Fix bugs regarding loading

Loading tasks was not performing as expected causing the loading to
fail and file to reset.

This commit:
- Fixes the bug in the loading method
- Fixes an exception
- Add title to the GUI window

WilkinsAng committed:
Add User Guide and Product Screenshot

WilkinsAng committed:
Update tests and fix spelling mistakes

WilkinsAng committed:
Fix uncaught bug in prioritise command

Code crashes when we try to set priority of an empty task list.
Date parses incorrectly for Smoke tester.

Let's:
- Check and handle the exception of empty list
- Force users to use Locale en-sg

WilkinsAng committed:
Update user guide with more important information

WilkinsAng committed:
Update user guide for formatting

WilkinsAng committed:
Update user guide with important information

WilkinsAng committed:
Update user guide to show auto sorting
